kuma/static/js/components/codepen-iex/codepen-iex.js

```diff
@@ -0,0 +1,21 @@
+(function() {
+    'use strict';
+    var sampleCodeContainer = document.querySelector('.codepen-iex');
+
+    /* only execute the rest of the code if there
+       is a `sampleCodeContainer` on the page */
+    if (sampleCodeContainer) {
+        var tryItButton = document.createElement('button');
+        tryItButton.classList.add('tryit');
+        tryItButton.textContent = gettext('Try it live!');
+        sampleCodeContainer.appendChild(tryItButton);
+
+        // when the button is clicked
+        tryItButton.addEventListener('click', function() {
+            // call codepen to load the editor
+            window.__CPEmbed('.codepen-iex');
```

Comment:
I can't find any docs for this __CPEmbed() function. Do you know if you can pass the DOM element directly to it instead of passing the selector as a string? If so, then we'd be able to support more than one example on a page by doing querySelectorAll, looping through the array and setting up the event handler to embed that one particular element in the query result.  I suppose we may never need that, but it would be cool.  

Another approach would be to check for an id attribute on the container element and use that id as the selector instead of a class selector if it exists.

Reply:
A couple of things here. First, we have agreed that there will never(I know never is a long time :)) have more than one of these on a page, so this should be fine.

At the moment the only documentation is here:
https://blog.codepen.io/documentation/prefill-embeds/

and more specifically here:
https://blog.codepen.io/documentation/prefill-embeds/#enhance-pre-blocks-into-editable-embeds-on-click-or-on-any-other-event-3

We could ask Chris directly though if this is possible, and it is worth doing so if this scenario seems likely. @wbamberg Thoughts?

Reply:
> First, we have agreed that there will never(I know never is a long time :)) have more than one of these on a page, so this should be fine.

It's not a requirement for now, but I think we might very well want to have > 1 of these on a page, at some point in the future.

Reply:
`window.__CPEmbed(".codepen-later");` is definitely a supported API, at the link @schalkneethling pointed to there. 

If you had more than 1 on a page, you could use different classes for each potentially? So they wouldn't be all triggered at once?

Reply:
Thanks for the feedback @chriscoyier - I agree that we should use different classes or, the clicked button needs to be context-aware so it knows which parent element to pass into the `__CPEmbed` function on click.